homeassistant/components/tplink/switch.py

```diff
@@ -38,9 +39,48 @@ async def async_setup_entry(
     else:
         entities.append(SmartPlugSwitch(device, coordinator))
 
+    entities.append(SmartPlugLedSwitch(device, coordinator))
+
     async_add_entities(entities)
 
 
+class SmartPlugLedSwitch(CoordinatedTPLinkEntity, SwitchEntity):
+    """Representation of switch for the LED of a TPLink Smart Plug."""
+
+    coordinator: TPLinkDataUpdateCoordinator
+
+    def __init__(
+        self, device: SmartDevice, coordinator: TPLinkDataUpdateCoordinator
+    ) -> None:
+        """Initialize the LED switch."""
+        super().__init__(device, coordinator)
+
+        self._attr_entity_category = ENTITY_CATEGORY_CONFIG
+        self._attr_name = f"{device.alias} (LED)"
+        self._attr_unique_id = f"{legacy_device_id(device)}_led"
+        self._attr_icon = "mdi:led-on"
```

Comment:
If you set a custom icon, you should set it for off as well

```suggestion


    @property
    def icon(self) -> str:
        """Return the icon for the LED."""
        return "mdi:led-on" if self.device.led else "mdi:led-off"
```